### ISSUE TITLE
Eureka InstanceInfo.Status and application Health.Status

### ISSUE THREAD

**Issue by jmnarloch** [spring-cloud/spring-cloud-netflix#508]
I'd noticed recently after hitting the /health endpoint of an application in status "DOWN", that this information does not propagate to Eureka, which will always be displayed with "UP" state (in version 1.0.3). This is not entirely perfect, especially if the application is down due to some external service like database connection or JMS broker being down. Even registering a health check that will always return DOWN does not change this, example:

```
@Bean
public HealthIndicator healthIndicator() {
     return () -> new Health.Builder().down().build();
}
```

I have made a Eureka HealthCheckHandler implementation that will propagate the health status from the application to Eureka: https://gist.github.com/jmnarloch/308dcaca18d3c2518663

I had to move the HealthIndicator retrieval to afterPropertiesSet to brake the cyclic bean dependencies between the health indicators.

On other hand the current Spring Boot policy "all or nothing" policy might be a bit restrictive for some use cases so I wonder whether to propose this as pull request or not. What do you thing?

**Comment by jmnarloch** [comment]
I've prepared a pull request for this issue: https://github.com/spring-cloud/spring-cloud-netflix/pull/509

I was able to test it in action (by that I mean I run it in actual system) and it works preaty sweet.

The only limitations that I see:

* No mapping for custom health statuses in Eureka, your custom application status becomes UNKNOWN in Eureka.
* In requires from you to become extra safe about the health indicators, for instance if your AMQP Turbine connection is down, despite not being crucial for the system to run, it will propagate to Eureka and to the clients that won't be sending traffic to this particular node. I can see couple of more examples of not critical services, like for instance cache, for which the health indicator will put the instance into DOWN state. So maybe a more fine grained control at some point in future could be considered as a plus.
